src/main/java/com/iota/iri/service/milestone/impl/MilestoneServiceImpl.java

```diff
@@ -175,9 +175,9 @@ public MilestoneValidity validateMilestone(TransactionViewModel transactionViewM
         }
 
         try {
-            if (MilestoneViewModel.get(tangle, milestoneIndex) != null) {
-                // Already validated.
-                return VALID;
+            MilestoneViewModel existingMilestone = MilestoneViewModel.get(tangle, milestoneIndex);
+            if(existingMilestone != null){
+                return existingMilestone.getHash().equals(transactionViewModel.getHash()) ? VALID : INVALID;
```

Comment:
Just that you know for future code,
I usually like to use `Objects#equals` to handle the `null` case

In this case I'd rather have a `NullPointerException` so we can find the bug early on :-)
So it is all good